arbnode/execution/tx_pre_checker.go

```diff
@@ -170,7 +171,7 @@ func PreCheckTx(bc *core.BlockChain, chainConfig *params.ChainConfig, header *ty
 				oldHeader = previousHeader
 				blocksTraversed++
 			}
-			if oldHeader != header {
+			if headerreader.HeadersEqual(oldHeader, header) {
```

Comment:
This condition is inverted which caused CI to fail. It should be `if !headerreader.HeadersEqual`